src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php

```diff
@@ -0,0 +1,182 @@
+        ) {
+            return $quoteTransfer;
+        }
+
+        $unzerCustomerTransfer = $this
+            ->unzerQuoteExpanderMapper
+            ->mapQuoteTransferToUnzerCustomerTransfer($quoteTransfer, new UnzerCustomerTransfer());
+
+        $unzerCustomerTransfer = $this->customerAdapter->createCustomer($unzerCustomerTransfer);
+        $quoteTransfer->getPayment()->getUnzerPayment()->setCustomer($unzerCustomerTransfer);
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    protected function createUnzerPayment(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        $paymentResourceTransfer = $this->extractPaymentResourceFromQuote($quoteTransfer);
+        $paymentSelection = $quoteTransfer->getPayment()->getPaymentSelection();
+
+        $unzerPaymentTransfer = new UnzerPaymentTransfer();
+        $unzerPaymentTransfer->setIsMarketplace($this->unzerConfig->isPaymentMethodMarketplaceReady($paymentSelection));
+        $unzerPaymentTransfer->setIsAuthorizable($this->unzerConfig->isPaymentAuthorizeRequired($paymentSelection));
+        $unzerPaymentTransfer->setPaymentResource($paymentResourceTransfer);
+
+        $quoteTransfer->getPayment()->setUnzerPayment($unzerPaymentTransfer);
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    protected function expandQuoteItemsWithUnzerParticipants(QuoteTransfer $quoteTransfer): QuoteTransfer
+    {
+        foreach ($quoteTransfer->getItems() as $itemTransfer) {
+            $this->setParticipantId($itemTransfer);
+        }
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ItemTransfer $itemTransfer
+     *
+     * @return \Generated\Shared\Transfer\ItemTransfer
+     */
+    protected function setParticipantId(ItemTransfer $itemTransfer): ItemTransfer
+    {
+        if (empty($itemTransfer->getMerchantReference())) {
+            return $itemTransfer;
+        }
+
+        $merchantUnzerParticipant = $this->unzerReader->getMerchantUnzerByMerchantReference($itemTransfer->getMerchantReference());
+        if ($merchantUnzerParticipant->getMerchantId() === null) {
+            return $itemTransfer;
+        }
+
+        return $itemTransfer->setUnzerParticipantId($merchantUnzerParticipant->getParticipantId());
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerPaymentResourceTransfer|null
+     */
+    protected function extractPaymentResourceFromQuote(QuoteTransfer $quoteTransfer): ?UnzerPaymentResourceTransfer
+    {
+        if ($quoteTransfer->getPayment()->getUnzerPayment()->getPaymentResource() !== null) {
```

Comment:
I guess that `$quoteTransfer->getPayment()` is deprecated maybe is better to check `getPayments` too.

Reply:
It used to save UnzerPaymentTransfer order info only